dask/dataframe/multi.py

```diff
@@ -419,6 +420,233 @@ def merge(left, right, how='inner', on=None, left_on=None, right_on=None,
                          indicator=indicator)
 
 
+###############################################################
+# ASOF Join
+###############################################################
```

Comment:
Given the amount of logic here, I think it would be fine to move this to a separate file, if you wanted (either way is fine).

Reply:
I prefer it in the `multi.py` file. It's only ~200 lines just like `merge` and `concat` are.